### DISCUSSION TITLE
Widgets location is not consistent

### DISCUSSION THREAD

**Discussion by kmurari103** [iTwin/itwinjs-core#2637] [Q&A]
Hi all,
I am using **"@itwin/web-viewer-react": "^1.3.1",**  package to create a web viewer app where i want to place some widgets at the top right just beside Tree View widget and some widgets just beside the Properties widget. I used End section to place widgets after Properties which works fine when i run app for the first time. But these widgets get dislocated once i re-run the app. If i clear the cache and browser history then it gets correct but again re-run means mis-aligned widgets . I also tried assigning priority to widgets but no luck. My question is - do i need to to clear browser history & cache everytime i re-run the app, shouldn't it take the section=End defined in the code to place the widgets?
Note: All other packages are latest one like "@bentley/ui-abstract": "^2.19.20", & others...
Here is the snippet how i'm trying this stuff: 

    if (stageId === "DefaultFrontstage" && stageUsage === StageUsage.General && location === StagePanelLocation.Right && section === StagePanelSection.Start) {
      widgets.push(
        {
          id: "smartDeviceListWidget",
          label: "Smart Devices",
          priority: 110,
          getWidgetContent: () => {
            return <span>This is devices</span>
          }
        }
      );
    }

    if (stageId === "DefaultFrontstage" && stageUsage === StageUsage.General && location === StagePanelLocation.Right && section === StagePanelSection.End) {
      const smartDeviceTypesWidget: AbstractWidgetProps = {
        id: "smartDeviceTypesWidget",
        label: "Device Types",
        priority: 120,
        defaultState: WidgetState.Closed,
        getWidgetContent: () => {
          return <span>This is type</span>
        }
      }

      const smartDeviceInfoWidget: AbstractWidgetProps = {
        id: "smartDeviceInfoWidget",
        label: "More Info",
        priority: 130,
        defaultState: WidgetState.Closed,
        getWidgetContent: () => {
          return <span>This is info</span>
        }
      }

      widgets.push(smartDeviceTypesWidget);
      widgets.push(smartDeviceInfoWidget);
    }
    return widgets;

After re-running app, i get widgets as this:
![image](https://user-images.githubusercontent.com/27068008/140031258-eae35594-1098-4069-96cf-afdfee4981a1.png)

**Comment by bsteinbk** [comment]
Do you run multiple itwin-viewer based apps using the same localhost url?  If so this could be confusing the saved state of the widgets.  By default the widget state is stored in browser local storage which stores it by url. Then we use the FrontstageId as a key to look up the last saved state of the stage. Since the itwin-view uses "DefaultFrontstage" all itwin-view based apps running on local host could be corrupting the saved state for the stage.

**Reply by kmurari103** [reply]
I don't run multiple viewer apps using same localhost url, This happens only if i make some code changes and hit save so the app reloads in browser and widgets get misplaced..
